packages/aws-cdk-lib/aws-sqs/lib/queue.ts

```diff
@@ -331,13 +385,33 @@ export class Queue extends QueueBase {
 
     validateProps(props);
 
+    if (props.redriveAllowPolicy) {
+      const { redrivePermission, sourceQueues } = props.redriveAllowPolicy;
+      if (redrivePermission === RedrivePermission.BY_QUEUE) {
+        if (!sourceQueues || sourceQueues.length === 0) {
+          throw new Error('At least one source queue must be specified when RedrivePermission is set to \'byQueue\'');
+        }
+        if (sourceQueues && sourceQueues.length > 10) {
+          throw new Error('Up to 10 sourceQueues can be specified. Set RedrivePermission to \'allowAll\' to specify more');
+        }
+      } else if (redrivePermission && sourceQueues) {
+        throw new Error('sourceQueues cannot be configured when RedrivePermission is set to \'allowAll\' or \'denyAll\'');
+      }
+    }
+
     const redrivePolicy = props.deadLetterQueue
       ? {
         deadLetterTargetArn: props.deadLetterQueue.queue.queueArn,
         maxReceiveCount: props.deadLetterQueue.maxReceiveCount,
       }
       : undefined;
 
+    const redriveAllowPolicy = props.redriveAllowPolicy ? {
```

Comment:
I understood this logic from reading the docstrings above, but in general this ternary condition is quite confusing for people who look at this piece of code. Would be nice to have a comment here explaining it.